static/app/components/core/layout/layer.tsx

```diff
@@ -0,0 +1,215 @@
+import {type DO_NOT_USE_ChonkTheme} from '@emotion/react';
+import styled from '@emotion/styled';
+
+import {chonkFor, debossedBackground} from 'sentry/components/core/chonk';
+import {
+  Container,
+  type ContainerElement,
+  type ContainerProps,
+} from 'sentry/components/core/layout/container';
+import {rc, type Responsive} from 'sentry/components/core/layout/styles';
+import {isChonkTheme} from 'sentry/utils/theme/withChonk';
+
+const slabElevation = {
+  md: '2px',
+  sm: '2px',
+  xs: '1px',
+} as const;
+
+const slabHoverElevation = '1px';
+interface SlabLayoutProps {
+  size?: NonNullable<Responsive<'sm' | 'md' | 'xs'>>;
+  variant?: 'accent' | 'warning' | 'danger';
+}
+
+type SlabProps<T extends ContainerElement = 'div'> = ContainerProps<T> & SlabLayoutProps;
+
+export function Slab<T extends ContainerElement = 'div'>(props: SlabProps<T>) {
+  const {variant, border, radius = 'md', size = 'md', ...rest} = props;
+
+  return (
+    <SlabContainer size={size} variant={variant} radius={radius}>
+      <Container
+        border={variant ?? 'primary'}
+        radius={radius}
+        background="primary"
+        {...rest}
+      />
+    </SlabContainer>
+  );
+}
+
+function getSlabContainerTheme(
+  variant: SlabLayoutProps['variant'],
+  theme: DO_NOT_USE_ChonkTheme
+) {
+  switch (variant) {
+    case 'accent':
+      return {
+        surface: theme.colors.blue400,
+        background: chonkFor(theme, theme.colors.blue400),
+      };
+    case 'warning':
+      return {
+        surface: theme.colors.chonk.yellow400,
+        background: chonkFor(theme, theme.colors.chonk.yellow400),
+      };
+    case 'danger':
+      return {
+        surface: theme.colors.chonk.red400,
+        background: chonkFor(theme, theme.colors.chonk.red400),
+      };
+    default:
+      return {
+        surface: theme.colors.surface500,
+        background: theme.colors.surface100,
+      };
+  }
+}
+
+const SlabContainer = styled(Container)<{
+  size: SlabLayoutProps['size'];
+  variant: SlabLayoutProps['variant'];
+}>`
+  background-color: ${p =>
+    isChonkTheme(p.theme)
+      ? getSlabContainerTheme(p.variant, p.theme).background
+      : p.theme.tokens.background.primary};
+
+  display: ${p => (isChonkTheme(p.theme) ? undefined : 'contents')};
+
+  > * {
+    ${p =>
+      rc(
+        'transform',
+        p.size,
+        p.theme,
+        (value, _breakpoint, _theme) => `translateY(-${slabElevation[value]})`
+      )};
+    transition: transform ${p => p.theme.motion.snap.fast};
+  }
+
+  &:hover {
+    > * {
+      ${p =>
+        rc(
+          'transform',
+          p.size,
+          p.theme,
+          (value, _breakpoint, _theme) =>
+            `translateY(calc(-${slabElevation[value]} - ${slabHoverElevation}))`
+        )};
+    }
+  }
+
+  &:active {
+    > * {
+      transform: translateY(0);
+    }
+  }
+
+  &:has(> *[aria-disabled='true']),
+  &[aria-disabled='true'] {
+    > * {
+      transform: translateY(0);
+    }
+  }
```

Comment:
Slightly concerned that this API is too magical/implicit. Would not be obvious to me to pass `aria-disabled` on this or on a child.

Reply:
Agree, I think I will look for a better approach